src/test/java/org/prebid/server/handler/NotificationEventHandlerTest.java

```diff
@@ -0,0 +1,238 @@
+package org.prebid.server.handler;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import io.netty.util.AsciiString;
+import io.vertx.core.MultiMap;
+import io.vertx.core.buffer.Buffer;
+import io.vertx.core.http.CaseInsensitiveHeaders;
+import io.vertx.core.http.HttpServerRequest;
+import io.vertx.core.http.HttpServerResponse;
+import io.vertx.ext.web.RoutingContext;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnit;
+import org.mockito.junit.MockitoRule;
+import org.prebid.server.VertxTest;
+import org.prebid.server.analytics.AnalyticsReporter;
+import org.prebid.server.analytics.model.NotificationEvent;
+import org.prebid.server.auction.model.Tuple2;
+import org.prebid.server.proto.request.EventNotificationRequest;
+import org.prebid.server.util.ResourceUtil;
+
+import java.io.IOException;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatNullPointerException;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyInt;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.Mockito.verify;
+
+public class NotificationEventHandlerTest extends VertxTest {
+
+    @Rule
+    public final MockitoRule mockitoRule = MockitoJUnit.rule();
+
+    @Mock
+    private RoutingContext routingContext;
+
+    @Mock
+    private HttpServerRequest httpRequest;
+
+    @Mock
+    private HttpServerResponse httpResponse;
+
+    @Mock
+    private AnalyticsReporter analyticsReporter;
+
+    private NotificationEventHandler notificationHandler;
+
+    @Before
+    public void setUp() {
+        given(routingContext.request()).willReturn(httpRequest);
+        given(routingContext.response()).willReturn(httpResponse);
+        given(httpRequest.params()).willReturn(MultiMap.caseInsensitiveMultiMap());
+        given(httpRequest.headers()).willReturn(new CaseInsensitiveHeaders());
+        given(httpResponse.putHeader(any(CharSequence.class), any(CharSequence.class))).willReturn(httpResponse);
+        given(httpResponse.setStatusCode(anyInt())).willReturn(httpResponse);
+        notificationHandler = NotificationEventHandler.create(analyticsReporter);
+    }
+
+    @Test
+    public void shouldThrowIllegalExceptionWhenReporterIsNull() {
```

Comment:
I think we stopped writing this kind of tests and gradually removing them from where they still exist.